src/core/surface/call.h

```diff
@@ -152,17 +151,17 @@ void grpc_call_context_set(grpc_call *call, grpc_context_index elem,
 void *grpc_call_context_get(grpc_call *call, grpc_context_index elem);
 
 #define GRPC_CALL_LOG_BATCH(sev, call, ops, nops, tag) \
-  if (grpc_trace_batch) grpc_call_log_batch(sev, call, ops, nops, tag)
+  if (grpc_surface_trace) grpc_call_log_batch(sev, call, ops, nops, tag)
```

Comment:
What advantage do you see in having two ways to log arguments passed to
grpc_call_start_batch?

On Thu, Aug 20, 2015 at 11:36 AM Masood Malekghassemi <
notifications@github.com> wrote:

> In src/core/surface/call.h
> <https://github.com/grpc/grpc/pull/3006#discussion_r37565211>:
>
> > @@ -152,17 +151,17 @@ void grpc_call_context_set(grpc_call *call, grpc_context_index elem,
> >  void *grpc_call_context_get(grpc_call *call, grpc_context_index elem);
> >
> >  #define GRPC_CALL_LOG_BATCH(sev, call, ops, nops, tag) \
> > -  if (grpc_trace_batch) grpc_call_log_batch(sev, call, ops, nops, tag)
> > +  if (grpc_surface_trace) grpc_call_log_batch(sev, call, ops, nops, tag)
>
> Should it...? I thought we wanted them separated? In particular, API
> tracing only traces entry into the surface, while surface tracing traces
> the values of operations being passed into and events being passed out from
> the surface layer. If they were merged, I'd also then want to add to API
> tracing a capture of return values, which seems a bit much.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc/pull/3006/files#r37565211>.
>